sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessorOptions.cs

```diff
@@ -0,0 +1,171 @@
+        /// The set of options to use for configuring the connection to the Service Bus entities.
+        /// </summary>
+        internal ServiceBusClientOptions _connectionOptions = new ServiceBusClientOptions();
+
+        /// <summary>
+        /// The set of options to govern retry behavior and try timeouts.
+        /// </summary>
+        internal ServiceBusRetryOptions _retryOptions = new ServiceBusRetryOptions();
+
+        /// <summary>
+        /// The number of messages that will be eagerly requested from Queues or Subscriptions and queued locally without regard to
+        /// whether a processing is currently active, intended to help maximize throughput by allowing the receiver to receive
+        /// from a local cache rather than waiting on a service request.
+        /// </summary>
+        public int PrefetchCount
+        {
+            get
+            {
+                return _prefetchCount;
+            }
+            set
+            {
+                if (value < 0)
+                {
+                    throw Fx.Exception.ArgumentOutOfRange(nameof(PrefetchCount), value, "Value cannot be less than 0.");
+                }
+                _prefetchCount = value;
+            }
+        }
+        private int _prefetchCount = 0;
+
+        /// <summary>
+        /// The <see cref="ReceiveMode"/> used to specify how messages are received. Defaults to PeekLock mode.
+        /// </summary>
+        public ReceiveMode ReceiveMode { get; set; } = ReceiveMode.PeekLock;
+
+        /// <summary>
+        /// Gets or sets the options used for configuring the connection to the Service Bus entities.
+        /// </summary>
+        internal ServiceBusClientOptions ConnectionOptions
+        {
+            get => _connectionOptions;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(ConnectionOptions));
+                _connectionOptions = value;
+            }
+        }
+
+        /// <summary>
+        /// The set of options to use for determining whether a failed operation should be retried and,
+        /// if so, the amount of time to wait between retry attempts.  These options also control the
+        /// amount of time allowed for receiving messages and other interactions with the Service Bus service.
+        /// </summary>
+        public ServiceBusRetryOptions RetryOptions
+        {
+            get => _retryOptions;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(RetryOptions));
+                _retryOptions = value;
+            }
+        }
+
+        /// <summary>Gets or sets a value that indicates whether the message-pump should call
+        /// Receiver.CompleteAsync() on messages after the callback has completed processing.</summary>
+        /// <value>true to complete the message processing automatically on successful execution of the operation; otherwise, false.</value>
+        public bool AutoComplete { get; set; }
```

Comment:
Don't we need default for `AutoComplete`?

Reply:
Yeah default should be false, but better to set explicitly.

Reply:
Can we update the comment saying that the default value is false?

Reply:
I will explicitly set it to false.

Reply:
Default for AutoComplete for true in Track 1 though....

Reply:
Ah so it is. I think this will be one that we should discuss with Service team about changing.

Reply:
I can make it True for now though..